docs(README): Update README structure

sergiojimcos opened the pull request:
**What's new?**
Update README.md structure including:
- **Table of Contents**.
- **New Cover**.
- **Better explanation in Setup content**
- **Better organisation**

**FIXES**
#4671

julien commented on README.md:
```diff
@@ -1,24 +1,29 @@
-The pillar to give colors and structure to Clay, where is the css and examples of HTML markups of the components.
-
-### [Clay Components](./packages)
+Clay is Liferay's web implementation of [Lexicon Experience Language](https://liferay.design/lexicon). Built with Bootstrap as a foundation, it's comprised of HTML, CSS, and JS.
```
```suggestion
Clay is Liferay's web implementation of the [Lexicon Experience Language](https://liferay.design/lexicon). Built with Bootstrap as a foundation, it's comprised of HTML, CSS, and JS.
```

julien commented on README.md:
```diff
@@ -31,6 +36,16 @@ Contains the source code and documentation powering [clayui.com](https://clayui.
+
+Clay repository follows the monorepo approach, all the major components that are the pillars of Clay are here in this repository.
+
+1. ### [Clay Css](./packages/clay-css): The pillar to give colors and structure to Clay, where is the css and examples of HTML markups of the components.
```
```suggestion
1. **[Clay Css]**(./packages/clay-css): The pillar to give colors and structure to Clay, where is the css and examples of HTML markups of the components.
```
Reason: I don't think we need to use an `h2` here because it looks "big", bold should be enough

julien commented on README.md:
```diff
@@ -31,6 +36,16 @@ Contains the source code and documentation powering [clayui.com](https://clayui.
+
+2. ### [Clay React Components](./packages): All components that follow clay-css marking and Lexicon use cases, developed on [React.js](http://reactjs.org)
+
+3. ### [Clayui.com](./clayui.com): Contains the source code and documentation powering [clayui.com](https://clayui.com/).
```
Same for this one

julien commented on README.md:
```diff
@@ -42,29 +57,23 @@ You can find the Clay documentation on the [site](https://clayui.com/docs).
+## Installation
 
-Feel free to open up problems or send pull requests. We will always be looking at these problems and we will be responding whenever possible.
+Clay is available as the `clay` package in npm for components and for css like `clay-css`. It is also available on a [CDN](https://cdn.jsdelivr.net/npm/clay/lib/js/clay.js) for components and for css, [check](https://cdn.jsdelivr.net/npm/clay/lib/css/atlas.css). But if you just want the separate components, it's also available in the npm as [clay-](https://www.npmjs.com/search?q=clay-).
```
The `clay` package in npm, is an older version of Clay as far as I can remember. 
Maybe we don't need to mention this (I wouldn't). Also, and just so you know and since this isn't easy to understand as someone "new" on the team: 

- In clay `2.x`, we used Metal.js and Soy and published the components with the following naming convention: `clay-COMPONENT_NAME`, for example: `clay-button`
- In clay `3.x`, we now use React (and TypeScript) and packages are published with the following naming convention: `@clayui/COMPONENT_NAME`, for example: `@clayui/button`

sergiojimcos replied:
If this point (Installation) explains `clay` npm package and it is an old version that someone new doesn't understand, I have two choices:

- Delete **Installation** point because it explains several concepts that are not easy to understand.
- In **Migration Guides** point I can add a small section with some information about clay `2.x` and clay `3.x`.

What do you think @julien?

julien replied:
I would go for 

> Delete Installation point because it explains several concepts that are not easy to understand.

Simply because no one is going to understand :rofl: 

Clay is available as "various" packages, there's actually one package per component, although that might change in the near future (see #4191 when you have time) 

Since this is the 3.x branch, I'd avoid mentioning anything about 2.x anyway, since it's just there for maintenance purposes or security fixes.

On top of that, anyone familiar with npm/React should be able to see very easily what the name (in the npm registry) of a component is by looking at the corresponding package.json file in the component's directory.

julien commented on README.md:
```diff
@@ -42,29 +57,23 @@ You can find the Clay documentation on the [site](https://clayui.com/docs).
+-   #### **Packages**
 
-#### Packages
+To contribute to Clay React Components, you can try to run locally to test your changes:
```
I'm not sure you need the "bold" there

julien commented on README.md:
```diff
@@ -42,29 +57,23 @@ You can find the Clay documentation on the [site](https://clayui.com/docs).
-#### Packages
+To contribute to Clay React Components, you can try to run locally to test your changes:
 
 1. Install NodeJS >= [v6.11.0](http://nodejs.org/dist/v6.11.0/), if you don't have it yet.
```
I would suggest, installing the latest "LTS" version, that way we don't need to update this every 6 weeks

sergiojimcos replied:
I'll change the Node version to 16.x, it seems that we are updated 😄

julien replied:
@sergiojimcos, I was actually saying that you shouldn't mention any specific version: clay should work with whatever LTS version is available. Don't worry I've fetched the branch and will push more stuff on top of your changes.

julien commented on README.md:
```diff
@@ -110,11 +119,11 @@ yarn test
 yarn run jest --u
```
`yarn run jest -u` is only needed to run the tests and update the snapshots.

julien commented on README.md:
````diff
@@ -110,11 +119,11 @@ yarn test
 ```
 
-#### clayui.com
+-   #### **Clayui.com**
````
I'm not sure you need this to be capitalized since it's an actual URL, and I don't think it needs to be in bold, since you're already using `####`.

julien commented on README.md:
```diff
@@ -110,11 +119,11 @@ yarn test
 To contribute to the documentation and the site in general, you can try to run locally to test your changes:
 
-1. Move to the site folder
+1. Move to the site folder:
```
```suggestion
Navigate to the site directory and install the dependencies
```

julien commented on README.md:
````diff
@@ -126,24 +135,32 @@ cd clayui.com
 ```
 
-If you want to test in a production environment so that you do not take risks of inconsistencies, in the root folder:
+> If you want to test in a production environment so that you do not take risks of inconsistencies, in the root folder:
````
> If you want to test in a production environment so that you do not take risks of inconsistencies, issue the following command in the root directory:

I prefer using "directory" over "folder", because that's usually how they are defined in operating systems and documentation

julien commented on README.md:
```diff
@@ -126,24 +135,32 @@ cd clayui.com
 
+## Contributing
+
+Feel free to open up problems or send pull requests. We will always be looking at these problems and we will be responding whenever possible.
```
Great addition, I'd just change it to 

```suggestion
Feel free to create issues or submit pull requests. Clay is actively maintained and your contributions are always welcome.
```

julien reviewed (commented):
@sergiojimcos pretty good changes, I left a few comments. If you need help, please let me know

sergiojimcos commented on README.md:
```diff
@@ -1,24 +1,29 @@
-## Packages
-
-Clay repository follows the monorepo approach, all the major components that are the pillars of Clay are here in this repository.
+</h1>
```
Yeah I've understood it. I agree with you, it's better to determine first a ´<div>´ instead of a child element 👍

julien replied:
Great

matuzalemsteles commented:
Well, I agree with @julien's additions, seems like a good job here @sergiojimcos on better organizing the information.

julien commented:
@sergiojimcos I just updated everything in https://github.com/liferay/clay/pull/4675/commits/6baffe4ab9db0eba13557c243e6e69c6377aead6 and squashed everything in 1 commit. Please tell me what you think when you get a chance. Thanks!

sergiojimcos commented:
> @sergiojimcos I just updated everything in [6baffe4](https://github.com/liferay/clay/commit/6baffe4ab9db0eba13557c243e6e69c6377aead6) and squashed everything in 1 commit. Please tell me what you think when you get a chance. Thanks!

LGFM! To be perfectionist, I'm going to add clayui.com in the Table of Contents 😄

julien commented:
> To be perfectionist, I'm going to add clayui.com in the Table of Contents smile

Good move @sergiojimcos !

sergiojimcos commented:
Hi @matuzalemsteles! README update is finished. You can review it!

matuzalemsteles reviewed (approved):
it's much better ❤️!
